### DISCUSSION TITLE
About maintaining parallel with Upstream yt-dlp

### DISCUSSION THREAD

**Discussion by bachig26** [kannagi0303/yt-dlp-gui#20] [General]
is the gui maintained in parallel with the yt-dlp latest release version?

or is it kept at outdated version?

**Comment by kannagi0303** [comment]
The short answer is: `not sure`.

First, as I mentioned in the project README.

Couldn't find an open source or free downloader suitable for my use, so I created this project, so I mainly update it according to my own needs.

If some users also like this way of using the GUI, I don't rule out adding some handy little features and updates for them (maybe new features to support yt-dlp)

However, this is a `personal project`, no major updates can be made quickly, and if it is a requirement that greatly exceeds the original capabilities of yt-dlp, I will not be able to add this feature.

**Reply by bachig26** [reply]
No, I'm not asking about implementing new features additional to yt-dlp.

My doubt was about the GUI being maintained in relation to the yt-dlp release version, like will it work properly with newer yt-dlp versions or will it able to support every functionality that yt-dlp gives that we can use with GUI. This was my doubts.

**Reply by kannagi0303** [reply]
GUI just acts as a middleman
Mainly to replace command line input.
So as long as yt-dlp does not change the names and usage of parameters, any version of yt-dlp can be used, even youtube-dl, they have most of the common parameters
You can think of a GUI as a parameter generator

Of course, I will only select the parameters commonly used by most people to make the corresponding GUI, and I will not make it for every function of yt-dlp. If you need to use all the functions, the most direct way is to use the command line. Enter parameters manually
